src/en/credentials.md

```diff
@@ -0,0 +1,134 @@
+Title: Juju credentials
+
+# Cloud credentials
+
+In order to access your cloud (with the exception of [LXD][lxd]), Juju will
+need to know how to authenticate itself. We use the term “credential” to 
+describe the tokens or keys or secrets needed to spin up resources on clouds.
+
+
+Currently, Juju can use one of three ways to get your credentials for a cloud:
+
+ - Scanning existing credentials (e.g. environment variables, .novarc files)
+ - Importing a user-provided [YAML-formatted][yaml] file.
+ - Entering credentials interactively on the command-line
+
+These methods are explained in more detail below.
+
+### Scanning existing credentials
+
+Some cloud providers (e.g. AWS, Openstack) have commandline tools which rely on 
+environment variables being used to store credentials. If these are in use on 
+your system already, or you choose to define them 
+([there is extra info here][env]), Juju can import them.
+
+For example, AWS uses the following environment variables (among others):
+
+**AWS_ACCESS_KEY_ID**
+
+**AWS_SECRET_ACCESS_KEY**
+
+If these are already set in your shell (you can echo $AWS_ACCESS_KEY_ID to test)
+they can be used by Juju.
+
+To scan your system for credentials Juju can use, run the command:
+
+```bash
+juju autoload-credentials
+```
+
+This will will interactively ask you whether to store each set of credentials
+it finds. Note that this is a 'snapshot' of those stored values - Juju will not 
+notice if they change in future.
+
+### Adding credentials via the commandline
+
+You can add credentials by running the command:
+
+```bash
+juju add-credential <cloud>
+```
+Juju will then interactively ask for the information it needs. This may vary 
+according to the cloud you are using, but will typically look something like
+this:
+
+```no-highlight
+juju add-credential aws 
+credential name: carol
+select auth-type [userpass, oauth, etc]: userpass
+enter username: cjones
+enter password: *******
+```
+
+Once you have supplied all the information, the credential will be added.
+
+At present, you will need to manually set this to be the default:
+
+```bash
+juju set-default-credential <cloud> <credential>
+```
+
+If you are having difficulty finding the correct credential information for 
+your cloud, please see the [help pages][help].
+
+### Adding credentials from a YAMl file.
+
+You can also specify a YAML format source file for the credentials. The source 
+file would be similar to:
+
+```yaml
+credentials:
+  aws:
+    default-credential: bob
+    default-region: us-east-1
```

Comment:
yes. So, again, just in case - this example is a) the one in the spec and b) one I know to work. I completely understand what you are saying, that some bits are ignored etc. However, if I change this, I will have to test it again. So, again, if you have a known to work example, please supply